website/content/docs/concepts/security/permissions.mdx

```diff
@@ -36,61 +116,121 @@ level action infers being granted all subactions. Thus, if a grant conveys
+discover auth methods and then authenticate to them), returning all
+configuration information for all scopes and auth methods felt like it was
+publicly disclosing more information than users might be desire.
 
-Every role assigns grants within a specific scope: either the scope in which the
-role exists, or a scope that is a child of the scope in which the role exists.
-This is controlled by the role's "grant scope ID".
+At the same time, this was not an ideal solution for two reasons:
 
-When a request is made, the scope in which to discover grants is either provided
-by the client (if against specific collection types) or is looked up using the
-resource's ID. This scope ID, along with the user's ID and the IDs of the groups
-the user belongs to, controls which roles are fetched to provide grants for the
-request.
+1. There is no one-size-fits-all security policy, and what we thought were
+   reasonable defaults may not work in all situations
 
-A role provides grants for a request if the grant scope ID matches the request's
-scope ID and one or more of the following are true:
+2. It made the scopes and auth methods listing behavior work differently from
+   any other `list` action, which is not ideal
 
-- The user's ID is contained in the principal IDs set on the role
-- A group the user belongs to is contained in the principal IDs set on the role
-- The user is logged in and the `u_auth` user is contained in the principal IDs set on the role
-- The role contains the `u_anon` user in the in the principal IDs set on the role
+As a result, we decided to approach this problem the same way we normally try to
+within Boundary: set resonable defaults, but give the administrator ultimate
+control.
+
+The resulting behavior is as follows: `output_fields` references the field names
+of the JSON object response, composes just like other parts of the RBAC model,
+and are action-specific. That is, a grant like:
+
+`id=*;type=auth-methods;actions=list,no-op;output_fields=scope_id,name,description`
+
+will, if all by itself, result in those three identified output fields applying
+to `list` (and `no-op`) actions for all auth methods in the scope. Thus when
+performing a `list`, each item returned will have only those three fields
+populated. Any other actions (like `read` or `update`) are unaffected and will
+use defaults.
+
+A grant like:
+
+`id=*;type=auth-methods;output_fields=id`
+
+will, if all by itself, result in _any_ action against auth methods in the scope
+having only `id` returned.
+
+However, if both of the above grants are included, since grants are composable
+the final set of output fields will be `id,scope_id,name,description` for `list`
+(and `no-op`) actions, and `id` for all other actions.
+
+If, after the grants are composed for a given request, none of the grants
+contain `output_fields` definitions (or if they do not apply to the
+resource/action in question), the defaults are used. These defaults are
```

Comment:
The text in the parenthesis is a little confusing IMO.  If the grant doesn't apply to the resource/action in question they would not be part of the grants composed for the given request, right? If so then the parenthasis is redundant and could probably be removed, if not, then maybe this can be reworded to be clearer?

Reply:
I reworded it, thanks!